### PR TITLE
upgrade to .NET 5

### PR DESCRIPTION
**Pull request by bospoort**
## Purpose
Upgraded the version of all projects to .NET 5.0

## Does this introduce a breaking change?

```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe:
Upgrade .NET version to .NET 5.0 
``` 

* Test the code

```
Build all
Run unit tests
```

## Other Information
No additional changes in the source had to be made.

### CODE REVIEW

File: src/Blazor.EventGridViewer.Core/Blazor.EventGridViewer.Core.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.1</TargetFramework>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
Comment (michaeldeongreen) [inline src/Blazor.EventGridViewer.Core/Blazor.EventGridViewer.Core.csproj:4]: @bospoort, this should be net5.0 like the others.  So you can verify yourself:
- Upgrade Visual Studio to the latest version (16.9.4)
- Right click on the .EventGridViewer..Core project => Properties => Target Framework => .Net 5.0

And it should change the value.

File: src/Blazor.EventGridViewer.Unit.Tests/Blazor.EventGridViewer.Unit.Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
Comment (michaeldeongreen) [inline src/Blazor.EventGridViewer.Unit.Tests/Blazor.EventGridViewer.Unit.Tests.csproj:4]: See previous comments.

**Review by michaeldeongreen — CHANGES_REQUESTED** [review]

File: src/Blazor.EventGridViewer.Unit.Tests/Blazor.EventGridViewer.Unit.Tests.csproj (hunk 2 above)
Comment (bospoort) [reply src/Blazor.EventGridViewer.Unit.Tests/Blazor.EventGridViewer.Unit.Tests.csproj]: Done

File: src/Blazor.EventGridViewer.Core/Blazor.EventGridViewer.Core.csproj (hunk 1 above)
Comment (bospoort) [reply src/Blazor.EventGridViewer.Core/Blazor.EventGridViewer.Core.csproj]: Updated. Thanks for coaching me through this Mike!

**Comment by bospoort** [comment]
Mike, I also updated the workflow, as the build action failed there with the old version. 

Sorry for not creating a branch. Rusty

**Comment by michaeldeongreen** [comment]
> Mike, I also updated the workflow, as the build action failed there with the old version.
> 
> Sorry for not creating a branch. Rusty

No Problem.

Can you delete [PR#16](https://github.com/Azure-Samples/eventgrid-viewer-blazor/pull/16) and add the changes in this PR?  All .Net 5 upgrades should go in the same PR.

**Comment by bospoort** [comment]
Closed PR16. This PR now also has the git action update to .NET 5.0 version

**Comment by michaeldeongreen** [comment]
@bospoort, looks like there is another change that needs to be made.  I deployed the app and the .NET version (azure portal => web app => Configuration => General settings tab) defaulted to *ASP.NET V4.8*.

The [ARM](https://github.com/Azure-Samples/eventgrid-viewer-blazor/blob/main/infrastructure/arm/azuredeploy.json) template will need to be modified.
- Add a new variable in the variables section of the ARM Template, maybe...netFrameworkVersion
- Set the value to "v5.0"
- Set the correct property in the ARM Template for the sites.  See last post [here](https://stackoverflow.com/questions/64942145/arm-template-to-create-a-net-5-app-service).  [Here](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites?tabs=json) is a full breakdown of the **microsoft.web/sites** section of an ARM Template for reference.

Let me know if you need any assistance.

**Comment by bospoort** [comment]
Working on it. Should be done by Monday. If that is OK.

**Comment by michaeldeongreen** [comment]
> Working on it. Should be done by Monday. If that is OK. 

Definitely no hurry, there is plenty of time.

**Comment by bospoort** [comment]
K. Fixed. Sort of.
- I kept deploying from the wrong ARM template initially (when hitting the "Deploy to Azure" button).
- the post mentioned didn't fix it for me. ARM validation did not like "value". Changing it to default Value "fixed" it.

**Comment by michaeldeongreen** [comment]
> K. Fixed. Sort of. - I kept deploying from the wrong ARM template initially (when hitting the "Deploy to Azure" button). - the post mentioned didn't fix it for me. ARM validation did not like "value". Changing it to default Value "fixed" it.

Sounds good.  Let's chat Monday.  The files in azure-pipelines need to be changed and also, parameter vs variable for the .net version.

**Comment by bospoort** [comment]
Changed from parameter to variable. (Note there are other values hardcoded (webSocketsEnabled, sku capacity), so there was no compelling reason to split this out in a variable).

**Review by michaeldeongreen — APPROVED** [review]
Test changes with both No-Auth and Auth.  LGTM!

**Pull request merged**